diff --git a/pysetup/generate_specs.py b/pysetup/generate_specs.py
index 59b12b15b..ef07a2fc8 100644
--- a/pysetup/generate_specs.py
+++ b/pysetup/generate_specs.py
@@ -1,6 +1,7 @@
 import argparse
 import ast
 import copy
+import json
 import sys
 from collections import OrderedDict
 from collections.abc import Sequence
@@ -23,6 +24,8 @@ from pysetup.md_to_spec import MarkdownToSpec
 from pysetup.spec_builders import spec_builders
 from pysetup.typing import BuildTarget, SpecObject  # type: ignore[attr-defined]
 
+pyspec = {}
+
 
 def get_spec(
     file_name: Path,
@@ -90,6 +93,14 @@ def build_spec(
     spec_object = all_specs[0]
     for value in all_specs[1:]:
         spec_object = combine_spec_objects(spec_object, value)
+
+    if preset_name not in pyspec:
+        pyspec[preset_name] = {}
+    if fork not in pyspec[preset_name]:
+        pyspec[preset_name][fork] = {}
+    for key, value in spec_object._asdict().items():
+        pyspec[preset_name][fork][key] = value.copy()
+
     spec_object = finalized_spec_object(spec_object)
 
     class_objects = {**spec_object.ssz_objects, **spec_object.dataclasses}
@@ -268,6 +279,10 @@ def generate_fork_specs(
     if verbose:
         print(f"  Wrote: {init_file}")
 
+    # Write pyspec dictionary to JSON file
+    with open("pyspec.json", "w") as file:
+        json.dump(pyspec, file)
+
 
 def main() -> int:
     """Main entry point for the spec generation script."""
